### PR TITLE
EOS-14276 Invalid Principal error during bucket policy validation

### PR DESCRIPTION
**Pull request by shalakadharap**

### CODE REVIEW

File: auth/server/src/main/java/com/seagates3/policy/BucketPolicyAuthorizer.java [hunk 1]
```diff
@@ -124,7 +125,9 @@ ServerResponse authorizeOperation(
     policyString = policyString.replace(
         "CanonicalUser",
         "Service");  // TODO:temporary solution till we implement parser
-    Policy existingPolicy = Policy.fromJson(policyString);
+    PolicyReaderOptions readerOptions = new PolicyReaderOptions();
+    readerOptions.setStripAwsPrincipalIdHyphensEnabled(false);
```
Comment (amitwac2608) [inline auth/server/src/main/java/com/seagates3/policy/BucketPolicyAuthorizer.java:129]: are there any other special characters which we need to take care of ?

**Review by amitwac2608 — CHANGES_REQUESTED** [review]
seems good. Can you also please add test for this ?

**Review by bkirunge7 — APPROVED** [review]

**Comment by shalakadharap** [comment]
> seems good. Can you also please add test for this ?

Already added .PLs re-check

File: auth/server/src/main/java/com/seagates3/policy/BucketPolicyAuthorizer.java (hunk 1 above)
Comment (amitwac2608) [reply auth/server/src/main/java/com/seagates3/policy/BucketPolicyAuthorizer.java:129]: No.

File: st/clitests/policy_spec.py [hunk 2]
```diff
@@ -100,6 +100,22 @@
...
+
+AwsTest('Aws can create bucket').create_bucket("usernametestbucket").execute_test().command_is_successful()
+user_args = {}
+user_args['UserName'] = "lyve_-iam101303"
+test_msg = "Create User lyve_-iam101303"
+user1_response_pattern = "UserId = [\w-]*, ARN = [\S]*, Path = /$"
+result = AuthTest(test_msg).create_user(**user_args).execute_test()
+result.command_should_match_pattern(user1_response_pattern)
+AwsTest("Aws can put policy on bucket").put_bucket_policy("usernametestbucket",user_name_testing).execute_test().command_is_successful()
+
```
Comment (amitwac2608) [inline st/clitests/policy_spec.py:116]: please clean the bucket after test.

Comment (amitwac2608) [reply st/clitests/policy_spec.py:116]: AwsTest('Delete Bucket "seagatebucket"').delete_bucket("seagatebucket")\
   .execute_test().command_is_successful()

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server/167/
build success

File: st/clitests/policy_spec.py (hunk 2 above)
Comment (shalakadharap) [reply st/clitests/policy_spec.py:119]: done.tnks!

**Review by nileshgovande — APPROVED** [review]

**Comment by nileshgovande** [comment]
Merged since Jenkins pass and ST added

**Pull request merged**